### PR TITLE
bindinfo, planner: fix sql bind when sql has symbol list (#13889)

### LINKED ISSUE #13871: sql bind will not be hit if the sql is similar as `select * from x limit 0, 1`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```mysql
create table t(a int, b int, INDEX ia (a), INDEX ib (b));
insert into t value(1, 1);

create global binding for select a, b, sleep(3) from t where a = 1 limit 0, 1 using select a, b, sleep(3) from t use index (ib) where a = 1 limit 0, 1;
```

Then run,
```mysql
select a, b, sleep(3) from t where a = 1 limit 0, 1;
```

2. What did you expect to see?
During the sleeping, show the execution plan in another client, but it still uses index `ia`
```mysql
mysql> show processlist;
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------+
| Id   | User | Host      | db   | Command | Time | State | Info                                                |
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------+
|    1 | root | 127.0.0.1 | test | Query   |    0 | 2     | select a, b, sleep(3) from t where a = 1 limit 0, 1 |
|    2 | root | 127.0.0.1 | test | Query   |    0 | 2     | show processlist                                    |
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------+
2 rows in set (0.00 sec)

mysql> explain for connection 1;
+--------------------------+-------+-----------+---------------------------------------------------------------+
| id                       | count | task      | operator info                                                 |
+--------------------------+-------+-----------+---------------------------------------------------------------+
| Projection_7             | 1.00  | root      | Column#1, Column#2, sleep(3)                                  |
| └─IndexLookUp_18         | 1.00  | root      | limit embedded(offset:0, count:1)                             |
|   ├─Limit_17             | 1.00  | cop[tikv] | offset:0, count:1                                             |
|   │ └─IndexScan_15       | 1.00  | cop[tikv] | table:t, index:a, range:[1,1], keep order:false, stats:pseudo |
|   └─TableScan_16         | 1.00  | cop[tikv] | table:t, keep order:false, stats:pseudo                       |
+--------------------------+-------+-----------+---------------------------------------------------------------+
5 rows in set (0.00 sec)
```

3. What did you see instead?

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

### PR DESCRIPTION
**Pull request by SunRunAway**
cherry pick #13889 

### What problem does this PR solve? 

Fix #13871, related PR https://github.com/pingcap/parser/pull/656

### What is changed and how it works?

Introduce `parser.DigestNormalized` to avoiding twice normalized.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - None

Side effects

 - None

Related changes

 - None

Release note

 - fix that SQL bind cannot be hit when a SQL has symbol list

### CODE REVIEW

**Comment by SunRunAway** [comment]
@zz-jason @eurekaka Please help merge.

File: bindinfo/handle.go [hunk 1]
```diff
@@ -16,11 +16,12 @@ package bindinfo
 import (
 	"context"
 	"fmt"
-	"go.uber.org/zap"
 	"sync"
 	"sync/atomic"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (zz-jason) [inline bindinfo/handle.go]: remove the empty line?

Comment (SunRunAway) [reply bindinfo/handle.go]: done.

**Pull request merged**